microcosm_sqlite/factories.py

```diff
@@ -11,11 +11,20 @@
 
 
 def on_connect_listener(use_foreign_keys):
-    def enforce_foreign_keys(connection, _):
+    def on_connect(dbapi_connection, _):
         if use_foreign_keys:
-            connection.execute("PRAGMA foreign_keys=ON")
+            dbapi_connection.execute("PRAGMA foreign_keys=ON")
 
-    return enforce_foreign_keys
+        # disable pysqlite's emitting of the BEGIN statement entirely,
+        # also stops it from emitting COMMIT before any DDL
+        # see: https://docs.sqlalchemy.org/en/latest/dialects/sqlite.html#serializable-isolation-savepoints-transactional-ddl  # noqa
+        dbapi_connection.isolation_level = None
+
+    return on_connect
+
+
+def on_begin_listener(connection):
+    connection.execute("BEGIN")
```

Comment:
I assume we execute `COMMIT` ourselves?

Reply:
Actually, according to https://docs.sqlalchemy.org/en/latest/dialects/sqlite.html?highlight=sqlite#serializable-isolation-savepoints-transactional-ddl only COMMITs before DDLs are not emitted, so `autocommit` session property should work as usual.